lib/datasource/pypi/index.ts

```diff
@@ -83,17 +84,33 @@ async function getDependency(
       }
     }
 
-    if (dep.info?.project_urls && !dependency.sourceUrl) {
+    if (dep.info?.project_urls) {
       for (const [name, projectUrl] of Object.entries(dep.info.project_urls)) {
         const lower = name.toLowerCase();
+
         if (
-          lower.startsWith('repo') ||
-          lower === 'code' ||
-          lower === 'source' ||
-          github_repo_pattern.exec(projectUrl)
+          !dependency.sourceUrl &&
+          (lower.startsWith('repo') ||
+            lower === 'code' ||
+            lower === 'source' ||
+            github_repo_pattern.exec(projectUrl))
         ) {
           dependency.sourceUrl = projectUrl;
-          break;
+        }
+
+        if (
+          [
+            'changelog',
+            'change log',
+            'changes',
+            'release notes',
+            'news',
+            "what's new",
+          ].includes(lower) ||
+          changelogFilenameRegex.exec(lower)
```

Comment:
```suggestion
          !dependency.changelogUrl &&
          ([
            'changelog',
            'change log',
            'changes',
            'release notes',
            'news',
            "what's new",
          ].includes(lower) ||
          changelogFilenameRegex.exec(lower))
```

Only assign if undefined